sdk/ai/azure-ai-inference/azure/ai/inference/aio/_patch.py

```diff
@@ -132,8 +132,158 @@ class ChatCompletionsClient(ChatCompletionsClientGenerated):
+         Default value is None.
+        :paramtype temperature: float
+        :keyword top_p: An alternative to sampling with temperature called nucleus sampling. This value
+         causes the
+         model to consider the results of tokens with the provided probability mass. As an example, a
+         value of 0.15 will cause only the tokens comprising the top 15% of probability mass to be
+         considered.
+         It is not recommended to modify temperature and top_p for the same completions request as the
+         interaction of these two settings is difficult to predict.
+         Supported range is [0, 1].
+         Default value is None.
+        :paramtype top_p: float
+        :keyword max_tokens: The maximum number of tokens to generate. Default value is None.
+        :paramtype max_tokens: int
+        :keyword response_format: An object specifying the format that the model must output. Used to
+         enable JSON mode. Known values are: "text" and "json_object". Default value is None.
+        :paramtype response_format: str or ~azure.ai.inference.models.ChatCompletionsResponseFormat
+        :keyword stop: A collection of textual sequences that will end completions generation. Default
+         value is None.
+        :paramtype stop: list[str]
+        :keyword stream: A value indicating whether chat completions should be streamed for this request.
+         Default value is False. If streaming is enabled, the response will be a StreamingChatCompletions.
+         Otherwise the response will be a ChatCompletions.
+        :paramtype stream: bool
+        :keyword tools: The available tool definitions that the chat completions request can use,
+         including caller-defined functions. Default value is None.
+        :paramtype tools: list[~azure.ai.inference.models.ChatCompletionsToolDefinition]
+        :keyword tool_choice: If specified, the model will configure which of the provided tools it can
+         use for the chat completions response. Is either a Union[str,
+         "_models.ChatCompletionsToolSelectionPreset"] type or a ChatCompletionsNamedToolSelection type.
+         Default value is None.
+        :paramtype tool_choice: str or ~azure.ai.inference.models.ChatCompletionsToolSelectionPreset or
+         ~azure.ai.inference.models.ChatCompletionsNamedToolSelection
+        :keyword seed: If specified, the system will make a best effort to sample deterministically
+         such that repeated requests with the
+         same seed and parameters should return the same result. Determinism is not guaranteed.
+         Default value is None.
+        :paramtype seed: int
+        :keyword model: ID of the specific AI model to use, if more than one model is available on the
+         endpoint. Default value is None.
+        :paramtype model: str
+        :return: this ChatCompletionsClient object, with defaults set.
+        :rtype: ~azure.ai.inference.aio.ChatCompletionsClient
+        """
```

Comment:
I noticed we have samples for each client using the with_defaults helper. A sample should showcase a champion scenario with the library, this feels more like something that could just be in the reference docs. I think adding an inline code usage sample to the docs here should be sufficient.

```suggestion
        :rtype: ~azure.ai.inference.aio.ChatCompletionsClient

        .. code-block:: python

            from azure.ai.inference.aio import ChatCompletionsClient
            from azure.ai.inference.models import SystemMessage, UserMessage
            from azure.core.credentials import AzureKeyCredential
        
            # Create a client with defaults
            client = ChatCompletionsClient(
                endpoint=endpoint,
                credential=AzureKeyCredential(key)
            ).with_defaults(
                temperature=0.5,
                max_tokens=1000
            )
        """
```

Reply:
@kristapratico thanks for the feedback. I'll schedule some time with you to discuss this, and potentially do some changes in a beta 4 PR. So far, I've been getting unprompted good feedback on the samples and how helpful they are.